tests/attestation_cert_apis/host/host.cpp

```diff
@@ -0,0 +1,145 @@
+#include <openenclave/internal/tests.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "tls_u.h"
+
+#define TEST_EC_KEY 0
+#define TEST_RSA_KEY 1
+#define SKIP_RETURN_CODE 2
+
+// This is the identity validation callback. An TLS connecting party (client or
+// server) can verify the passed in "identity" information to decide whether to
+// accept an connection reqest
+oe_result_t enclave_identity_verifier(oe_identity_t* identity, void* arg)
+{
+    oe_result_t result = OE_VERIFY_FAILED;
+
+    (void)arg;
+    OE_TRACE_INFO("enclave_identity_verifier is called with parsed report:\n");
+
+    // Check the enclave's security version
+    OE_TRACE_INFO(
+        "identity.security_version = %d\n", identity->security_version);
+    if (identity->security_version < 1)
+    {
+        OE_TRACE_ERROR(
+            "identity.security_version check failed (%d)\n",
+            identity->security_version);
+        goto done;
+    }
+
+    // Dump an enclave's unique ID, signer ID and Product ID. They are
+    // MRENCLAVE, MRSIGNER and ISVPRODID for SGX enclaves.  In a real scenario,
+    // custom id checking should be done here
+    OE_TRACE_INFO("identity->signer_id :\n");
+    for (int i = 0; i < OE_UNIQUE_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->signer_id[i]);
+
+    OE_TRACE_INFO("\nidentity->signer_id :\n");
+    for (int i = 0; i < OE_SIGNER_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->signer_id[i]);
+
+    OE_TRACE_INFO("\nidentity->product_id :\n");
+    for (int i = 0; i < OE_PRODUCT_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->product_id[i]);
+
+    result = OE_OK;
+done:
+    return result;
+}
+
+void run_test(oe_enclave_t* enclave, int test_type)
+{
+    oe_result_t result = OE_FAILURE;
+    oe_result_t ecall_result;
+    unsigned char* cert = NULL;
+    size_t cert_size = 0;
+
+    OE_TRACE_INFO(
+        "Host: get tls certificate signed with %s key from an enclave \n",
+        test_type == TEST_RSA_KEY ? "a RSA" : "an EC");
+    if (test_type == TEST_EC_KEY)
+    {
+        result = get_tls_cert_signed_with_ec_key(
+            enclave, &ecall_result, &cert, &cert_size);
+    }
+    else if (test_type == TEST_RSA_KEY)
+    {
+        result = get_tls_cert_signed_with_rsa_key(
+            enclave, &ecall_result, &cert, &cert_size);
+    }
+
+    if ((result != OE_OK) || (ecall_result != OE_OK))
+        oe_put_err(
+            "get_tls_cert_signed_with_%s_key() failed: result=%u",
+            test_type == TEST_RSA_KEY ? "rsa" : "ec",
+            result);
+
+    fflush(stdout);
+
+    {
+        // for testing purpose, output the whole cer in DER format
+        char filename[80];
+        FILE* file = NULL;
+
+        sprintf(
+            filename,
+            "./cert_%s.der",
+            test_type == TEST_RSA_KEY ? "rsa" : "ec");
+        OE_TRACE_INFO(
+            "Host: Log quote embedded certificate to file: [%s]\n", filename);
```

Comment:
I believe the term "quote" is SGX specific.  Use platform agnostic terms in trace output from non-SGX-specific code